index.js

```diff
@@ -19,6 +21,13 @@ class Bot extends EventEmitter {
          this.name = params.name;
 
          console.assert(params.token, 'token must be defined');
+
+         // Use a HTTP Proxy if defined in the environment. We need to accomodate both uppercase and lowercase environment definitions
```

Comment:
Just as sugestion, this could be refactored into an encapsulated proxy setup method or similar :)

Reply:
Done. As a bonus, defining a `proxy` param in the constructor will be respected before looking at the environment.